src/errors/http_status_errors.cr

```diff
@@ -0,0 +1,30 @@
+require "http"
+require "./exception"
+
+module DppmRestApi
+  abstract class HTTPStatusError < Exception
+    abstract def status_code
+  end
+
+  {% for status_code in HTTP::Status.constants %}
+  {% type_name = status_code.stringify.downcase.camelcase.id %}
+  # An exception for the {{status_code}} status
+  class {{type_name.id}} < HTTPStatusError
+    STATUS_CODE = HTTP::Status::{{status_code}}
+
+    def status_code
+      STATUS_CODE
+    end
+
+    def self.status_code
+      STATUS_CODE
+    end
```

Comment:
Why?

Reply:
Why not?

Reply:
Because one can already use `STATUS_CODE`. Unless you have something against uppercase letters?